OpenEdXMobile/src/main/java/org/edx/mobile/module/prefs/PrefBaseManager.kt

```diff
@@ -0,0 +1,215 @@
+
+    /**
+     * Puts given key-value pair to the Shared Preferences.
+     *
+     * @param key
+     * @param value - int
+     */
+    fun put(key: String?, value: Int) {
+        editor.putInt(key, value).apply()
+    }
+
+    /**
+     * Returns String value for the given key, null if no value is found.
+     *
+     * @param key
+     * @return String
+     */
+    fun getString(key: String): String? = sharedPreferences.getString(key, null)
+
+    /**
+     * Returns String value for the given key, null if no value is found.
+     *
+     * @param key
+     * @param defaultValue
+     * @return String
+     */
+    fun getString(key: String, defaultValue: String): String =
+        sharedPreferences.getString(key, defaultValue) ?: defaultValue
+
+    /**
+     * Returns boolean value for the given key, can set default value as well.
+     *
+     * @param key
+     * @param defaultValue
+     * @return boolean
+     */
+    fun getBoolean(key: String, defaultValue: Boolean): Boolean =
+        sharedPreferences.getBoolean(key, defaultValue)
+
+    /**
+     * Returns long value for the given key, -1 if no value is found.
+     *
+     * @param key
+     * @param defaultValue
+     * @return long
+     */
+    fun getLong(key: String, defaultValue: Long = -1): Long =
+        sharedPreferences.getLong(key, defaultValue)
+
+    /**
+     * Returns float value for the given key, defaultValue if no value is found.
+     *
+     * @param key
+     * @param defaultValue
+     * @return float
+     */
+    fun getFloat(key: String, defaultValue: Float = -1f): Float =
+        sharedPreferences.getFloat(key, defaultValue)
+
+    /**
+     * Returns int value for the given key, defaultValue if no value is found.
+     *
+     * @param key,default value
+     * @return int
+     */
+    fun getInt(key: String, defaultValue: Int = -1): Int =
+        sharedPreferences.getInt(key, defaultValue)
+
+    fun removeKey(key: String) = editor.remove(key).apply()
+
+    fun clear() = editor.clear().apply()
+
+    fun migrateData(oldPrefBaseManager: PrefBaseManager) {
+        oldPrefBaseManager.sharedPreferences.all.forEach {
+            when (it.value) {
+                is Int -> put(it.key, it.value as Int)
+                is Long -> put(it.key, it.value as Long)
+                is Float -> put(it.key, it.value as Float)
+                is Boolean -> put(it.key, it.value as Boolean)
+                is String -> put(it.key, it.value as String)
+            }
+            oldPrefBaseManager.removeKey(it.key)
+        }
+    }
+
+    /**
+     * Contains preference name constants. These must be unique.
+     */
+    object Pref {
+        const val LOGIN = "pref_login"
+        const val WIFI = "pref_wifi"
+        const val VIDEOS = "pref_videos"
+        const val APP_FEATURES = "app_features"
+        const val COURSE_CALENDAR_PREF = "course_calendar_pref"
+        const val APP_INFO = "pref_app_info"
+        const val USER_PREF = "pref_user"
```

Comment:
Rename `LOGIN` to `LOGIN_PREF` & reorder them Alphabetically:  (We can ignore Wifi, videos and calendar)

```
        const val APP_FEATURES = "app_features"
        const val APP_INFO = "pref_app_info"
        const val LOGIN_PREF = "pref_login"
        const val USER_PREF = "pref_user"
        const val WIFI = "pref_wifi"
        const val VIDEOS = "pref_videos"
        const val COURSE_CALENDAR_PREF = "course_calendar_pref"